spec/compiler/macro/macro_methods_spec.cr

```diff
@@ -1957,6 +1993,17 @@ module Crystal
           )) { int32 }
         end
 
+        it "type within another type from the top level with extra ::" do
+          assert_type(%(
+            class A
+              class B
+              end
+            end
+
+            {{ @type.has_constant?("A::::::B") ? 1 : 'f' }}
```

Comment:
Not sure if failing silently instead of raising a macro error is the best way to approach it. I think we should perform a full parsing (similar to `parse_type`) and require the result to be a `Path`. Put in another way, if these two methods also accept `Path` arguments, then passing a `StringLiteral` would be a shorthand for:

```crystal
{{
  path = parse_type("A::::::B")
  raise "..." unless path.is_a?(Path)
  @type.has_constant?(path)
}}
```

Reply:
I think invalid syntax should be an error.